### ISSUE TITLE
Display all executed process instance modifications

### ISSUE THREAD

**Issue by korthout** [camunda/camunda#18376]
### Description

Operate only shows the current state, the taken path, and whether or not any element instances have been terminated. With [Process Instance Modification](https://docs.camunda.io/docs/next/components/concepts/process-instance-modification/), I can activate, move, or terminate element instances. The results will be visible in the process instance's overview and instance history.

However, it's not clear in many cases that a modification was the cause of the events in the instance history and overview.

It would be beneficial to have:
- a list of all Process Instance Modifications applied to a process instance
- for each applied modification, a visual representation of the changes, or a before and after

We encountered that this would've been useful in root causing today's investigation of [a process instance](https://bru-2.operate.camunda.io/689a796f-7efa-487f-9e44-76ca3a98c77b/processes/4503599664803449) that unexpectedly had an extra token after a parallel gateway. This made us think we had a bug in the engine (see https://github.com/camunda/zeebe/issues/18375). However, one possible explanation would've been that somebody modified the process instance. 

Operate offers no way to see whether the instance has been modified. This would be valuable in many cases, such as traceability, troubleshooting, etc.

For similar reasons, this would also be useful for [Process Instance Migration](https://docs.camunda.io/docs/next/components/concepts/process-instance-migration/).

### User Story
Not filled in by me:

```Gherkin
Scenario: As an Operate user
    Given I am logged in
    When I select one item/tab in the toolbar
    Then this item should be shown in highlighted (in bold)
```
### Acceptance Criteria

- [ ]

### Definition of Ready - Checklist

- [ ] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] If documentation needs to be updated, an issue is created in the [camunda-docs](https://github.com/camunda/camunda-docs) repo, and the issue is added to our Operate project board.
- [ ] If HELM charts need to be updated, an issue is created in the [camunda-platform-heml](https://github.com/camunda/camunda-platform-helm) repo, and the issue is added to our Operate project board.
- [ ] Potential Security risks have been considered and added to the product-hub issue risk assessment
- [ ] Cross-team dependencies have been considered

Optional:
- [ ] Design input has been collected by the assignee
### :point_right: Handover Dev to QA 

- Handy resources: 
     BPMN/DMN models, plugins, scripts, REST API endpoints + example payload, etc : 
     
- Example projects:

- Versions to validate:
> Docker file : in case that it needed to be tested via docker share the version contain the fixed along with version of other services . 

- Release version ( in which version this fixed/feature will be released)

**Comment by johanwelgemoed** [comment]
@gastonpillet01  & @aleksander-dytko could you have a look and help determine next steps please :)

**Comment by aleksander-dytko** [comment]
@gastonpillet01 I belive this will be covered by Audit log of User Operations. Could you validate designs with @korthout ?

**Comment by gastonpillet01** [comment]
@korthout could you please take a look at this [first prototype for Audit log](https://www.figma.com/proto/EUyFWlMg03ZGvuvO1yMfa8/Audit-log?page-id=268%3A84795&node-id=470-208316&viewport=911%2C-1806%2C0.1&t=tMCMtYOLUOJBHoD2-1&scaling=min-zoom&starting-point-node-id=470%3A208316&show-proto-sidebar=1), and share your thoughts or any kind of feedback?

**Comment by korthout** [comment]
Thanks, @gastonpillet01 👏 Great work on this! And I'm happy to help review 🙇 

Things I really like:
- The Audit Log is a main tab 🚀 
- The Audit Log provides an overview with filtering options 🚀 
- Comments seem like a powerful feature (I hope the filtering supports fuzzy matching) 🚀 
- I managed to end up in the `Instance log` tab, which provides an excellent overview of everything that happened to my instance 🚀 
- I love that the `Instance log` shows who, when, and what 🚀 
- Did I already say I like the comments feature? 😄 

Things I don't like so much or don't understand:
1. When I click `Batch Move` in the Audit Log, I'm sent to a page where I think I see the operations themselves (there's success/failure), but they are named by their process, not the operation. It makes it unclear what I'm actually looking at.
2. I cannot see the details of the operations. What did the `Batch Move` operation do? What did it look like before? What did it look like after?
3. When I click `Batch retry` in the Audit Log, I see a different view than when I clicked `Batch Move`. It shows me a process instance overview rather than the operations overview I saw after clicking `Batch Move`. 
4. It's a bit hidden, but I could find a tooltip that shows up when hovering and tells me that two instances were retried. But which ones?
5. When I click `Instance modification` in the `Instance log`, I have the same ability to hover for a tooltip that tells me that two instances were added. Again, I'm left with the question, which ones? What happened to those instances? Perhaps the shown data is incorrect for the case, or I don't understand it.

So, to recap, I see some things that could be improved, but there is also a lot to love! 🚀 I've numbered the improvement points to facilitate individual discussion. Let me know if you have any questions.

**Comment by korthout** [comment]
Thanks @gastonpillet01 ❤️  I really appreciate it. No need for a meeting IMO, I can share my thoughts here.

1. Is now clear to me.
2. (and 4,5) I hope we can work to overcome technical challenges to improve the UX. I believe many users will have use cases where they need Operate to have the capability to show what happened during a modification, e.g. traceability, root-causing, etc
3. This looks great! Thanks

**Comment by korthout** [comment]
I had a look whether we could close this issue, but I suggest we keep it open.

The Operations Log now covers the main traceability need: process-instance modification and migration operations are recorded and can be viewed from an instance, including actor, timestamp, outcome, and operation type.

However, this issue also asks for the concrete change made by each modification—e.g., activated/terminated/moved element instances—and a visual or before/after representation. The current Operations Log does not appear to expose those modification instructions or affected elements.

I suggest we keep this issue open for that remaining requirement, or split it into a follow-up focused on modification/migration detail visualization and close this issue as partially delivered.
